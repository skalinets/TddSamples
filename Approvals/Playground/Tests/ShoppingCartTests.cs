using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NUnit.Framework;
using ApprovalTests;
using Playground.Core;
using ApprovalTests.Reporters;

namespace Playground.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class ShoppingCartTests
    {

        [Test]
        public void should_calculate_the_total_price_for_shopping_cart()
        {
            // do
            var shoppingCart = new ShoppingCart();
            shoppingCart.Add(new Product { Id = "iPad", Price = 500 });
            shoppingCart.Add(new Product { Id = "Mouse", Price = 20 });
            shoppingCart.Confirm();

            // verify
            Approvals.Approve(shoppingCart);
        }

    }
}
