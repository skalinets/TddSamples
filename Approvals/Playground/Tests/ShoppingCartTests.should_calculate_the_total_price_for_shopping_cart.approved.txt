Total products in cart: 2
Products: [Product: iPad Price: 500, Product: Mouse Price: 20]
Total price: 520